autonomy/cli/helpers/analyse.py

```diff
@@ -88,59 +94,199 @@ def run_dialogues_check(
         raise click.ClickException(str(e))
 
 
-def _load_agent_from_ipfs(agent: PublicId) -> AgentConfig:
+def _get_content_from_ipfs(package_id: PackageId, file: str) -> bytes:
+    """Read content from the IPFS registry."""
+    try:
+        ipfs_tool = IPFSTool()
+        return ipfs_tool.client.cat(
+            f"{package_id.package_hash}/{package_id.name}/{file}"
+        )
+    except Exception as e:
+        raise click.ClickException(
+            "Fetching content from the IPFS registry failed"
+            f"\n\tPackage: {package_id}\n\tFile: {file}\n\tError: {e}"
+        ) from e
+
+
+def _load_from_ipfs(package_id: PackageId) -> PackageConfiguration:
     """Load agent config from the IPFS hash."""
-    ipfs_tool = IPFSTool()
+    with tempfile.TemporaryDirectory() as package_dir:
+        config_file = PACKAGE_TYPE_TO_CONFIG_FILE[package_id.package_type.value]
+        config_content = _get_content_from_ipfs(package_id=package_id, file=config_file)
+        Path(package_dir, config_file).write_bytes(config_content)
+        return load_configuration_object(
+            package_type=package_id.package_type,
+            directory=Path(package_dir),
+            package_type_config_class=PACKAGE_TYPE_TO_CONFIG_CLASS,
+        )
+
+
+def _load_from_local(
+    package_id: PackageId, package_manager: PackageManagerV1
+) -> PackageConfiguration:
+    """Load from local repository."""
+
+    return load_configuration_object(
+        package_type=package_id.package_type,
+        directory=package_manager.package_path_from_package_id(package_id=package_id),
+        package_type_config_class=PACKAGE_TYPE_TO_CONFIG_CLASS,
+    )
 
-    with tempfile.TemporaryDirectory() as agent_dir:
-        agent_config_content = ipfs_tool.client.cat(
-            f"{agent.hash}/{agent.name}/{DEFAULT_AEA_CONFIG_FILE}"
+
+def _get_chained_abci_skill(
+    agent_config: AgentConfig,
+    package_manager: PackageManagerV1,
+    is_on_chain_check: bool = False,
+) -> Optional[SkillConfig]:
```

Comment:
@DavidMinarsch @Adamantios Method for finding the chained ABCI app skill for an agent